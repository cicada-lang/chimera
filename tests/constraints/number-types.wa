assert find _ {
  Number 1
}

assert not find _ {
  Number "A"
}

assert not find x {
  x = "cat"
  Number x
}

assert find x {
  x = 1
  Number x
}

assert not find x {
  Number x
  String x
}

// The inequality is subsumed by type constraint:

find x {
  x != "cat"
  Number x
}

find [x, y] {
  [x, y] != ["cat", 1]
  Number x
}

find x {
  Number x
  x != "cat"
}

find [x, y] {
  Number x
  [x, y] != ["cat", 1]
}

find x {
  Number z
  z = x
  x != "cat"
}
