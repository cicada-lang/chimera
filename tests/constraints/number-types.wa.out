[ _.0 with { Number _.0 } ]
[ [_.0, _.1] with { Number _.0 } ]
[ _.0 with { Number _.0 } ]
[ [_.0, _.1] with { Number _.0 } ]
[ _.0 ]
